kotskinds/multitype/quotedbool.go

```diff
@@ -0,0 +1,75 @@
+// Based on https://github.com/kubernetes/apimachinery/blob/455a99f/pkg/util/intstr/intstr.go
+
+package multitype
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+// QuotedBool is a string type that can also unmarshal raw yaml bools.
+//
+// +protobuf=true
+// +protobuf.options.(gogoproto.goproto_stringer)=false
+// +k8s:openapi-gen=true
+// +kubebuilder:validation:Type=QuotedBool
+type QuotedBool string
+
+// UnmarshalJSON implements the json.Unmarshaller interface.
+func (b *QuotedBool) UnmarshalJSON(value []byte) error {
+	trueValues := []string{"y", "Y", "yes", "Yes", "YES", "true", "True", "TRUE", "on", "On", "ON"}
+	falseValues := []string{"n", "N", "no", "No", "NO", "false", "False", "FALSE", "off", "Off", "OFF"}
+	for _, v := range trueValues {
+		if string(value) == v {
+			*b = "true"
+			return nil
+		}
+	}
+	for _, v := range falseValues {
+		if string(value) == v {
+			*b = "false"
+			return nil
+		}
+	}
+	var s string
+	if err := json.Unmarshal(value, &s); err != nil {
+		return errors.Wrapf(err, "unable to unmarshal %q as bool or string", string(value))
+	}
+	*b = QuotedBool(s)
+	return nil
+}
+
+// UnmarshalJSON implements the yaml.Unmarshaller interface.
+func (b *QuotedBool) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	boolTry := false
+	stringTry := ""
+	err := unmarshal(&boolTry)
```

Comment:
why dont you just use this logic in the function above as well?

Reply:
Because this is relying on a function passed in as a param, and UnmarshalJSON has a different signature?